spec.bs

```diff
@@ -8372,49 +8761,78 @@ To <dfn>batch and fetch trusted scoring signals</dfn> given a [=trusted scoring
+            trusted scoring signals authorization from a fetcher=] given |scriptFetcher|.
+          1. If |allowCrossOriginTrustedScoringSignalsFrom| does not [=list/contain=]
+            |fullSignalsUrl|'s [=url/origin=], set |fullSignalsUrl| to null.
+        1. Let |result| be failure.
+        1. If |fullSignalsUrl| is not null:
+          1. Let «|allTrustedScoringSignals|, <var ignore>ignored</var>, |scoringDataVersion|» be
+            the result of [=fetching trusted signals=] with |fullSignalsUrl|, |seller|,
+            |entriesToBatch|[0]'s [=trusted scoring signals request/policy container=], and false.
+          1. If |allTrustedScoringSignals| is an [=ordered map=]:
+            1. Set |result| to a new [=trusted scoring signals reply=]
+            1. Set |result|'s [=trusted scoring signals reply/all trusted scoring signals=] to
+              |allTrustedScoringSignals|.
+            1. Set |result|'s [=trusted scoring signals reply/data version=] to |scoringDataVersion|.
+        1. [=list/For each=] |entry| in |entriesToBatch|:
+          1. Set |entry|'s [=trusted scoring signals request/reply=] to |result|.
+    1. Otherwise:
+      1. Set |entriesToBatch| to |batcher|'s [=trusted scoring signals batcher/request map=][|key|].
+      1. [=map/Remove=] |key| from |batcher|'s [=trusted scoring signals batcher/request map=].
+      1. Let « |requestBody|, |renderUrlIdMap|, |context| » be the result of [=building batched
+        trusted key value scoring signals request body=] for |entriesToBatch|.
+      1. Let |baseUrl| be |entriesToBatch|[0]'s [=trusted scoring signals request/base url=].
       1. Let |seller| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller=].
       1. Let |scriptFetcher| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller
         script fetcher=].
-      1. Let |policyContainer| be |entriesToBatch|[0]'s [=trusted scoring signals request/policy
-        container=].
-      1. If |fullSignalsUrl|'s [=url/origin=] is not [=same origin=] with |seller| then:
+      1. If |baseUrl|'s [=url/origin=] is not [=same origin=] with |seller| then:
         1. Let |allowCrossOriginTrustedScoringSignalsFrom| be the result of [=wait for cross origin
           trusted scoring signals authorization from a fetcher=] given |scriptFetcher|.
         1. If |allowCrossOriginTrustedScoringSignalsFrom| does not [=list/contain=]
-          |fullSignalsUrl|'s [=url/origin=], set |fullSignalsUrl| to null.
+          |baseUrl|'s [=url/origin=], set |baseUrl| to null.
       1. Let |result| be failure.
-      1. If |fullSignalsUrl| is not null:
+      1. If |baseUrl| is not null:
         1. Let «|allTrustedScoringSignals|, <var ignore>ignored</var>, |scoringDataVersion|» be
-          the result of [=fetching trusted signals=] with |fullSignalsUrl|, |seller|,
-          |policyContainer|, and false.
+          the result of [=fetching trusted key value signals=] with |baseUrl|, |requestBody|,

```

Comment:
now `|requestBody|, |renderUrlIdMap|, |context|` can all be null, so you'd either update `[=fetching trusted key value signals=]`'s definition to allow these parameters to be null, or just check whether |requestBody| is null here, and only call the algorithm when it's not null.

Reply:
So in `fetch trusted key value signals`, I have:
```
To <dfn>fetch trusted key value signals</dfn> given a [=URL=] |url|, a [=byte sequence=] |body|, a [=string=]
|context|, an [=origin=] |scriptOrigin|, a [=policy container=] |policyContainer|, an |indexMap| whose keys
are [=strings=] and values are [=tuples=] consisting of an interger and an integer , and a [=boolean=]
|isBiddingSignal|:

1. If |body| is null, return « null, null, null ».
```
which I think is able to handle the null request body pass-in scenario.

Reply:
so in your definition, |context| is a string, but it can be null now, since your `[=building batched trusted key value scoring signals request body=]` returned null for all three variables. So you can either change the algorithm definition to say "a [=string=]-or-null |context|" (and for the other param as well), or let `[=building batched trusted key value scoring signals request body=]` return failure instead of three nulls, and only invoke `[=fetch trusted key value signals=]` when it's not failure, then |context| is always string.

Reply:
Done, add `-or-null` to type of |context| and |indexMap| in `fetch trusted key value signals`.

Reply:
(this is not critical, so can be in your follow up PR, but would be best to make the change. It's not hard to change, so probably do it in this PR makes sense. Up to you)

actually I think after line 8811 (build request body step), you can do:
```
1. Let |result| be failure.
1. If |requestBody| is not null:
  ... //all the following steps except the last step of the algorithm
  ...
1. [=list/For each=] |entry| in |entriesToBatch|: ...
```
This way, the params to fetch algorithm would not be null. And it's weird to execute all the following steps when |requestBody| is null, since we already know the request would fail. Also, if body is null, other two are null, and if body is not null, other two won't be null. The current -or-null params cannot express that.

Reply:
Gotcha, changed to checking if requestBody is not null and removed the check in fetch trusted key value signals.